src/internal/util/logger.ts

```diff
@@ -1,119 +1,286 @@
 import * as cluster from "cluster";
 import * as fs from "fs-extra";
-import * as stringify from "json-stringify-safe";
 import * as _ from "lodash";
+import * as logform from "logform";
 import * as os from "os";
 import * as p from "path";
 import * as serializeError from "serialize-error";
-import * as stripAnsi from "strip-ansi";
 import * as winston from "winston";
+import { Configuration } from "../../configuration";
 import * as context from "./cls";
 
-export let LoggingConfig = {
-    format: "logger",
+const winstonLogger = winston.createLogger({
+    level: "debug",
+    exitOnError: false,
+    silent: true,
+});
+
+/**
+ * Global logger instance
+ * By default all logging to this logger will be suppressed.
+ * Call configureLogging to set up console and file transports.
+ */
+export const logger: Logger = winstonLogger;
+
+// Save console log methods
+const GlobalConsoleMethods = {
+    error: console.error,
+    warn: console.warn,
+    info: console.info,
+    debug: console.debug,
+    log: console.log,
+    trace: console.trace,
 };
 
-export function setLogLevel(level: string) {
-    winstonLogger.transports.console.level = level;
+/**
+ * Constants for the logging format
+ */
+export enum LoggingFormat {
+    /**
+     * Print only the message provided to the logger methods
+     */
```

Comment:
If you want to condense this a bit, you can enter the one-line TypeDoc like this:

```typescript
/** Print only the message ... */
```